src/rust/src/x509/verify.rs

```diff
@@ -0,0 +1,105 @@
+// This file is dual licensed under the terms of the Apache License, Version
+// 2.0, and the BSD License. See the LICENSE file in the root of this repository
+// for complete details.pub(crate) struct PyCryptoOps {}
+
+use cryptography_x509::certificate::Certificate;
+use cryptography_x509_validation::ops::CryptoOps;
+
+use crate::error::CryptographyResult;
+
+use super::sign;
+
+pub(crate) struct PyCryptoOps {}
+
+impl CryptoOps for PyCryptoOps {
+    // NOTE: This "key" type also carries any error that might happen
+    // during key serialization/loading. This isn't ideal but also is
+    // not a significant issue, since its only use is in checking
+    // another certificate's signature (where an error means that the
+    // signature check fails trivially).
+    type Key = CryptographyResult<pyo3::Py<pyo3::PyAny>>;
+
+    fn public_key(&self, cert: &Certificate<'_>) -> Self::Key {
```

Comment:
Is there a reason not to have this return a result on the trait?

Reply:
> Is there a reason not to have this return a result on the trait?

No major reason -- conceptually it's an infallible operation so I figured it made sense to "bury" the result type here, but I can make the return type `Result<Self::Key>` instead if you'd prefer.

Reply:
I don't think I agree that it's conceptually infallible -- it's very possible for a cert to have a malformed public key!

Reply:
True -- I'm thinking too much in terms of typestates again 🙂 

I'll make this a `Result<...>` return.